aten/src/TH/generic/THTensorRandom.cpp

```diff
@@ -310,6 +314,10 @@ void THTensor_(multinomial)(THLongTensor *self, THGenerator *_generator, THTenso
                           THCleanup(THDoubleTensor_free(cum_dist); if (start_dim == 1) THTensor_(squeeze1d)(prob_dist, prob_dist, 0);),
                           2,
                           "invalid multinomial distribution (sum of probabilities <= 0)");
+    THArgCheckWithCleanup((n_categories - n_zeros >= n_sample),
+                          THCleanup(THDoubleTensor_free(cum_dist); if (start_dim == 1) THTensor_(squeeze1d)(prob_dist, prob_dist, 0);),
```

Comment:
Man, it will be very nice when we can switch this to C++ smart pointer refcounting :)

Reply:
I'm a little puzzled why there's a call to `squeeze1d` in the cleanup code.

Reply:
@ezyang  Is it doable in TH/THC? Can you point me to an example code? ;) Thanks!

Reply:
No, not yet, unless you want to port this entire kernel to ATen :(

Reply:
@ezyang We unsqueezed it inplace before. I guess it's mainly for keeping the input not changed. :P 
Btw I created #12499 to port it to Aten, will get to it maybe next week. :)